### PR TITLE
Make robolectric runtime dependency support specifying source path

### PR DESCRIPTION
**Pull request by raviagarwal7**

### CODE REVIEW

**Comment by styurin** [comment]
Can you provide an example how it should be used? We have integration tests for robolectric_test.

**Comment by raviagarwal7** [comment]
> Can you provide an example how it should be used? We have integration tests for robolectric_test.

so when providing source path it would be a genrule:
```
http_file(
    name = 'android-all-5.0.2_r3-robolectric-r0.jar',
    urls = [ 'mvn:org.robolectric:android-all:jar:5.0.2_r3-robolectric-r0' ],
    sha256 = '5e63d4c7f2c691afed648bf0675e0b0a76d19f0e23d93705f4faf9ed3b2734de',
)


http_file(
    name = 'android-all-8.1.0-robolectric-4611349.jar',
    urls = [ 'mvn:org.robolectric:android-all:jar:8.1.0-robolectric-4611349' ],
    sha256 = '01b3364bc0b4ca099929aa4d818529c7ec05e18b9e4c1e667f99955b16ae19b4',
)


targets = [
    ':android-all-8.1.0-robolectric-4611349.jar',
    ':android-all-5.0.2_r3-robolectric-r0.jar',
]


bash_list = (
    ["mkdir -p $OUT; "] +
    ['cp $(location {}) $OUT/{}; '.format(target, target[1:]) for target in targets]
)


genrule(
    name = 'robolectric_cache',
    bash = " ".join(bash_list),
    out = "cache",
    visibility = [ 'PUBLIC' ],
)
```

Now the robolectric_test rule will have `robolectricRuntimeDeps = '//:robolectric_cache` as the source path.

in case of path we would specify `robolectricRuntimeDeps = <output of robolectric_cache>`

We can now have a `robolectric_test` rule which uses this, but I am not sure how to set that up in the current `RobolectricTestRuleIntegrationTest.java` class. https://github.com/facebook/buck/blob/master/test/com/facebook/buck/android/RobolectricTestRuleIntegrationTest.java#L39

any ideas?

**Comment by styurin** [comment]
I recently updated robolectric integration tests to specify runtime dir in vm_args using a macro: 9a880c642cf20e25e5d36b7f0329ce4d6706c984. It looks the same except that you would need to use `robolectric_runtime_dependency` instead of `vm_aergs`.

**Comment by raviagarwal7** [comment]
> I recently updated robolectric integration tests to specify runtime dir in vm_args using a macro: [9a880c6](https://github.com/facebook/buck/commit/9a880c642cf20e25e5d36b7f0329ce4d6706c984). It looks the same except that you would need to use `robolectric_runtime_dependency` instead of `vm_aergs`.

yup, updated to use  `robolectric_runtime_dependency` instead of `vm_args`

File: test/com/facebook/buck/android/testdata/android_project/java/com/sample/lib/BUCK.fixture [hunk 1]
```diff
@@ -99,9 +99,7 @@ robolectric_test(
     visibility = [
         "PUBLIC",
     ],
-    vm_args = [
-        "-Drobolectric.dependency.dir=$(location buck//third-party/java/robolectric:robolectric_dir)",
-    ],
+    robolectric_runtime_dependency = "buck//third-party/java/robolectric:robolectric_dir",
```
[the remark this answers is not in the document]
Comment (styurin) [reply test/com/facebook/buck/android/testdata/android_project/java/com/sample/lib/BUCK.fixture]: I think we should provide both test cases, can you create another test for `robolectric_runtime_dependency`?

[the remark this answers is not in the document]
Comment (raviagarwal7) [reply test/com/facebook/buck/android/testdata/android_project/java/com/sample/lib/BUCK.fixture]: sure, updated the pr.

File: test/com/facebook/buck/android/testdata/android_project/java/com/sample/lib/BUCK.fixture [hunk 2]
```diff
@@ -113,3 +113,22 @@ robolectric_test(
         "//res/com/sample/top:top",
     ],
 )
+
+robolectric_test(
+    name = "test_robolectric_runtime_dep",
```
Comment (styurin) [inline test/com/facebook/buck/android/testdata/android_project/java/com/sample/lib/BUCK.fixture]: You need to add a test in `RobolectricTestRuleIntegrationTest` that uses this target.

**Comment by styurin** [comment]
Could you rebase? Looks like there a merge conflict.

**Comment by styurin** [comment]
Committed as 90d62021dd3af51e437ae28c4a06231dd252fa98. I tried to committed with a proper author, but looks like that information was lost somewhere.

**Pull request closed without merging**